File: inl1/src/inl1/RacingEvent.java
package inl1;

import java.util.Random;
import lab5.ColorTurtle;
import se.lth.cs.window.SimpleWindow;

public class RacingEvent {

	private RaceTrack track;
	private ColorTurtle[] t;
	
	/** Skapar ett lopp mellan sköldpaddorna t på
	 * kapplöpningsbanan track */
	public RacingEvent(RaceTrack track, ColorTurtle[] t) {
		this.track = track;
		this.t = t;
	}
	/** Startar loppet, och skriver ut vinnaren vid slutet */
	public void startRace(SimpleWindow w) {
		track.draw(w);
		Random rand = new Random();
		Boolean finished = false;
		String text = "";
		while(true) {
			for(ColorTurtle t1 : t)
				t1.forward(rand.nextInt(3));
			// Check if tie
			for(int i = 0; i < t.length; i++) {
				if(t[i].getY() < track.getFinish()) {
					finished = true;
					text += (i+1) + " ";
				}
			}
			if(finished) {
				w.moveTo(w.getWidth()/2 - 40, w.getHeight() / 2);
				w.writeText("Turtle " + text + "won!");
				return;
			}
			SimpleWindow.delay(10);
		}
	}
}

File: Scala/src/inl1/RacingEvent.scala
package inl1

import java.util.Random;
import lab4.ColorTurtle
import se.lth.cs.window.SimpleWindow

class RacingEvent (val track: RaceTrack, val turtle1: ColorTurtle, val turtle2: ColorTurtle) {
  def startRace(w: SimpleWindow) {
    track.draw(w)
    val rand = new Random()
    while(true) {
      turtle1.forward(rand.nextInt(3))
      turtle2.forward(rand.nextInt(3))
      if(turtle1.getY < track.getFinish || turtle2.getY < track.getFinish) {
        w.moveTo(w.getWidth/2 - 40, w.getHeight / 2)
        if(turtle1.getY < turtle2.getY)
          w.writeText("Turtle 1 won!")
        else
          w.writeText("Turtle 2 won!")
        return
      }
      Thread.sleep(10)
    }
  }
}